### PR TITLE
lvm2: fix musl patches

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

See commit messages.

This isn't enough to actually fix the Musl build, but it's as far as I was able to get.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 7c6f434c** [comment]
Looks fine from non-musl point of view.

File: pkgs/os-specific/linux/lvm2/default.nix [hunk 1]
```diff
@@ -67,23 +67,14 @@ stdenv.mkDerivation rec {
...
-
-  patches = lib.optionals stdenv.hostPlatform.isMusl [
-    (fetchpatch {
-      name = "fix-stdio-usage.patch";
-      url = "https://git.alpinelinux.org/aports/plain/main/lvm2/fix-stdio-usage.patch?h=3.7-stable&id=31bd4a8c2dc00ae79a821f6fe0ad2f23e1534f50";
-      sha256 = "0m6wr6qrvxqi2d2h054cnv974jq1v65lqxy05g1znz946ga73k3p";
-    })
+  patches = [
+    # Musl fixes from Alpine.
+    ./fix-stdio-usage.patch
```
Comment (ajs124) [inline pkgs/os-specific/linux/lvm2/default.nix:72]: Why apply this unconditionally? Without this, wouldn't this be a noop for the non-musl case?

Comment (alyssais) [reply pkgs/os-specific/linux/lvm2/default.nix:72]: Conditionally applied patches mean increased maintenance burden.  It's the reason these patches were able to bitrot in the first place.  This PR is to staging, where there's almost always a full stdenv rebuild anyway, so it doesn't matter at all whether it's a no-op.

Comment (ajs124) [reply pkgs/os-specific/linux/lvm2/default.nix:72]: Alright :+1:

I also meant to ask/say that if this was conditional, we could just go directly to master with it.

I've been trying to always test this when updating/modifying lvm2, but iirc it was mostly broken on master already and I don't use musl myself, so I didn't fix it. Sorry if I broke this.

Anyways, lets go ahead with this then. Is there a musl team or can I ping you if I run into problems with this in the future?

Comment (alyssais) [reply pkgs/os-specific/linux/lvm2/default.nix:72]: There's an exotic platform team that I'm in, maybe that's the one to use?

**Review by ajs124 — APPROVED** [review]
Can't run the `installer.lvm` test right now, because `cargo-c` is broken on `staging`, but the `lvm2` package still builds, which is good enough for me.

**Pull request merged**